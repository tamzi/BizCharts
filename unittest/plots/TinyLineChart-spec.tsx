import React from 'react';
import TinyLineChart from '../../src/plots/TinyLineChart';
import { render, cleanup } from '@testing-library/react';

const MOCK_DATA = [
    { year: '1991', value: 3 },
    { year: '1992', value: 4 },
    { year: '1993', value: 3.5 },
    { year: '1994', value: 5 },
    { year: '1995', value: 4.9 },
    { year: '1996', value: 6 },
    { year: '1997', value: 7 },
    { year: '1998', value: 9 },
    { year: '1999', value: 13 },
];


const basicCfg = {
    data: MOCK_DATA,
    width: 400,
    height: 400,
    xField: 'year',
    yField: 'value',
}


describe('Plots-TinyLineChart', () => {
    test('基础TinyLineChart', () => {
        let chart = null;
        render(<TinyLineChart
            {...basicCfg}
            size={10}
            onGetG2Instance={(c) => {
                chart = c;
            }}
        />);
        expect(chart.options.lineStyle.lineWidth).toBe(10);
        expect(chart.options.data[0]).toBe(3);
        expect(chart.options).toMatchSnapshot();
        cleanup();
    });
    test('TinyLineChart-guidLine-type', () => {
        let chart = null;
        render(<TinyLineChart
            {...basicCfg}
            smooth
            color="pink"
            lineStyle={{
                stroke: 'red' // 优先级高于color
            }}
            guideLine={[
                {
                    type: 'mean',
                    lineStyle: {
                        stroke: 'red',
                    },
                    text: {
                        position: 'start',
                        content: '平均值',
                        style: {
                            lineWidth: 2,
                        },
                    },
                },
            ]}
            onGetG2Instance={(c) => {
                chart = c;
            }}
        />);
        expect(chart.options).toMatchSnapshot();
        cleanup();
    });
    test('TinyLineChart-guidLine-start', () => {
        let chart = null;
        render(<TinyLineChart
            {...basicCfg}
            smooth
            color="pink"
            lineStyle={{
                stroke: 'red' // 优先级高于color
            }}
            guideLine={[
                {
                    start: ['min', 5],
                    end: ['max', 5],
                    lineStyle: {
                        stroke: 'red',
                    },
                    text: {
                        position: 'start',
                        content: '值:5',
                        style: {
                            lineWidth: 2,
                        },
                    },
                }
            ]}
            onGetG2Instance={(c) => {
                chart = c;
            }}
        />);
        expect(chart.options).toMatchSnapshot();
        cleanup();
    });

});
