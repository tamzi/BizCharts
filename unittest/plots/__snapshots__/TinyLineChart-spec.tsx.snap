// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plots-TinyLineChart TinyLineChart-guidLine-start 1`] = `
Object {
  "animation": true,
  "annotations": Array [
    Object {
      "end": Array [
        "max",
        5,
      ],
      "lineStyle": Object {
        "stroke": "red",
      },
      "start": Array [
        "min",
        5,
      ],
      "style": Object {
        "stroke": "red",
      },
      "text": Object {
        "content": "值:5",
        "position": "start",
        "style": Object {
          "lineWidth": 2,
        },
      },
      "type": "line",
    },
  ],
  "appendPadding": Array [
    10,
    5,
    10,
    10,
  ],
  "autoFit": true,
  "color": "pink",
  "data": Array [
    3,
    4,
    3.5,
    5,
    4.9,
    6,
    7,
    9,
    13,
  ],
  "guideLine": Array [
    Object {
      "end": Array [
        "max",
        5,
      ],
      "lineStyle": Object {
        "stroke": "red",
      },
      "start": Array [
        "min",
        5,
      ],
      "text": Object {
        "content": "值:5",
        "position": "start",
        "style": Object {
          "lineWidth": 2,
        },
      },
    },
  ],
  "height": 400,
  "lineStyle": Object {
    "stroke": "red",
  },
  "point": false,
  "smooth": true,
  "tooltip": false,
  "width": 400,
  "xField": "year",
  "yField": "value",
}
`;

exports[`Plots-TinyLineChart TinyLineChart-guidLine-type 1`] = `
Object {
  "animation": true,
  "annotations": Array [
    Object {
      "end": Array [
        "max",
        6.155555555555555,
      ],
      "lineStyle": Object {
        "stroke": "red",
      },
      "start": Array [
        "min",
        6.155555555555555,
      ],
      "style": Object {
        "stroke": "red",
      },
      "text": Object {
        "content": "平均值",
        "position": "start",
        "style": Object {
          "lineWidth": 2,
        },
      },
      "type": "line",
    },
  ],
  "appendPadding": Array [
    10,
    5,
    10,
    10,
  ],
  "autoFit": true,
  "color": "pink",
  "data": Array [
    3,
    4,
    3.5,
    5,
    4.9,
    6,
    7,
    9,
    13,
  ],
  "guideLine": Array [
    Object {
      "lineStyle": Object {
        "stroke": "red",
      },
      "text": Object {
        "content": "平均值",
        "position": "start",
        "style": Object {
          "lineWidth": 2,
        },
      },
      "type": "mean",
    },
  ],
  "height": 400,
  "lineStyle": Object {
    "stroke": "red",
  },
  "point": false,
  "smooth": true,
  "tooltip": false,
  "width": 400,
  "xField": "year",
  "yField": "value",
}
`;

exports[`Plots-TinyLineChart 基础TinyLineChart 1`] = `
Object {
  "animation": true,
  "appendPadding": Array [
    10,
    5,
    10,
    10,
  ],
  "autoFit": true,
  "data": Array [
    3,
    4,
    3.5,
    5,
    4.9,
    6,
    7,
    9,
    13,
  ],
  "height": 400,
  "lineStyle": Object {
    "lineWidth": 10,
  },
  "size": 10,
  "tooltip": false,
  "width": 400,
  "xField": "year",
  "yField": "value",
}
`;
